gadget/gadget.go

```diff
@@ -1013,3 +1013,29 @@ func PositionedVolumeFromGadget(gadgetRoot string) (*LaidOutVolume, error) {
 	}
 	return nil, fmt.Errorf("internal error in PositionedVolumeFromGadget: this line cannot be reached")
 }
+
+func flatten(path string, cfg interface{}, out map[string]interface{}) {
+	if cfgMap, ok := cfg.(map[string]interface{}); ok {
+		for k, v := range cfgMap {
+			p := k
+			if path != "" {
+				p = path + "." + k
+			}
+			flatten(p, v, out)
+		}
+	} else {
+		out[path] = cfg
+	}
+}
+
+// CoreDefaults returns default system (or core) configuration from gadget defaults.
+func CoreDefaults(gadgetDefaults map[string]map[string]interface{}, coreSnapID string) map[string]interface{} {
```

Comment:
not something you can solve here now, but I really want to avoid having to pass in the coreSnapID, I'm not even sure the places that will use this have any better access to it.

There are some UC20 not important TODOs  that would need something like naming.WellKnwonSnapdID("core"|"snapd"....)   I'm probably going to fix that which should help with this.

Reply:
Right, I was wondering about this aspect, but wasn't sure about the places this is going to be called from.

Reply:
I proposed https://github.com/snapcore/snapd/pull/8335 not the most urgent thing in itself, but seems it's needed by this code anyway